pyiceberg/table/__init__.py

```
@@ -807,7 +806,7 @@ def upsert(
             case_sensitive=case_sensitive,
         )
 
-        if branch is not None:
+        if branch in self.table_metadata.refs:
```

Here we silently ignore the branch if it doesn't exists. I think it would be better to raise a `ValueError`

---

ah I was thinking for upsert, we also might want to support staging commit as well so I remove the exception. 
Let me know what you think!